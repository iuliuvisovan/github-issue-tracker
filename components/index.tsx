import { Dimensions } from 'react-native';

const { width: vw, height: vh } = Dimensions.get('screen');

export { vw };
export { vh };

export { default as Button } from './button';
export { default as TextInput } from './text-input';
export { default as Text } from './text';
export { default as Color } from './color';
