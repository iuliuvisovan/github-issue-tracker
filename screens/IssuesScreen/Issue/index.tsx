import React from 'react';
import isDarkColor from 'is-dark-color';
import moment from 'moment';
import { View, Image, TouchableOpacity } from 'react-native';
import { Color, Text } from '../../../components';
import { IIssueItemProps } from '../../../types/issues';
import styles from './styles';
import { AntDesign, Feather } from '@expo/vector-icons';
import Header from './Header';

export default function Issue(props: IIssueItemProps) {
  const { issue } = props;

  const goToDetails = () => {
    props.navigation.push('IssueDetails', { issue });
  };

  return (
    <TouchableOpacity activeOpacity={0.8} onPress={goToDetails} style={styles.card}>
      <Header issue={issue} />
      <Text style={styles.title}>{issue.title}</Text>
      <View style={styles.labels}>
        {issue.labels.map((label) => (
          <View key={label.id} style={[styles.label, { backgroundColor: '#' + label.color }]}>
            <Text style={[styles.labelText, { color: isDarkColor('#' + label.color) ? Color.white : Color.steel }]}>
              {label.name.replace(/\:([a-z]+)\:/g, '')}
            </Text>
          </View>
        ))}
      </View>
    </TouchableOpacity>
  );
}
