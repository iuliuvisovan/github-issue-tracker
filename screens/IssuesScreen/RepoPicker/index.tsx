import React, { useState } from 'react';
import { View, TouchableOpacity } from 'react-native';
import { Button, Color, Text, TextInput } from '../../../components';
import styles from './styles';
import useCurrentPage from '../useCurrentPage';
import useIssues from '../../../hooks/useIssues';
import { AntDesign, Feather } from '@expo/vector-icons';

export default function RepoPicker() {
  const currentPageManager = useCurrentPage();
  const { organizationId: commitedOrganizationId, repoId: commitedRepoId, isPickerOpen } = currentPageManager.data;
  const { setIsPickerOpen, commitOrganizationId, commitRepoId } = currentPageManager.actions;

  const issuesManager = useIssues();
  const { getIssues } = issuesManager.actions;

  const [organizationId, setOrganizationId] = useState(commitedOrganizationId);
  const [repoId, setRepoId] = useState(commitedRepoId);

  const viewIssues = () => {
    commitOrganizationId(organizationId);
    commitRepoId(repoId);
    getIssues();
    setIsPickerOpen(false);
  };

  const cancel = () => {
    setIsPickerOpen(false);
    setRepoId(commitedRepoId);
    setRepoId(commitedOrganizationId);
  };

  const githubIcon = <AntDesign size={20} color={Color.blue} name="github" style={{ marginRight: 6 }} />;

  return isPickerOpen ? (
    <View style={styles.card}>
      <View style={styles.inputs}>
        <TextInput name="Organization" testID="organizationInput" value={organizationId} onChangeText={setOrganizationId} />
        <Text style={{ marginTop: 26, fontSize: 20, marginLeft: 2, color: Color.border }}>/</Text>
        <TextInput name="Repository" value={repoId} onChangeText={setRepoId} />
      </View>
      <View style={styles.buttonsWrapper}>
        <Button type="secondary" testID="cancelButton" text="Cancel" onPress={cancel} style={{ width: 130 }} />
        <Button type="quaternary" text="View issues" leftIcon={githubIcon} onPress={viewIssues} />
      </View>
    </View>
  ) : (
    <View style={[styles.spacedRow, { marginTop: 0 }]}>
      <Feather size={20} name="git-branch" color={Color.border} style={styles.filterIcon} />
      <Text style={styles.repoText}>{`${commitedOrganizationId} / ${commitedRepoId}`}</Text>
      <TouchableOpacity testID="expandPickerButton" onPress={() => setIsPickerOpen(true)} style={styles.editRepoButton}>
        <Feather size={14} name="edit-2" color={Color.blue} />
      </TouchableOpacity>
    </View>
  );
}
