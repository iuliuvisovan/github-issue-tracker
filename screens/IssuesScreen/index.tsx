import React, { useEffect, useState } from 'react';
import { AntDesign, Feather, FontAwesome } from '@expo/vector-icons';

import { ActionSheetIOS, ActivityIndicator, LayoutAnimation, View } from 'react-native';
import { FlatList, TouchableOpacity } from 'react-native-gesture-handler';

import { useDispatch, useSelector } from 'react-redux';
import { IApplicationState } from '../../redux';
import * as issueActions from '../../redux/actions/issues';
import * as bookmarkActions from '../../redux/actions/bookmarks';

import { Button, Color, TextInput, Text } from '../../components';
import Issue from './Issue';
import styles from './styles';
import { IIssuesScreenProps } from '../../types/navigation';
import { LinearGradient } from 'expo-linear-gradient';

export default function IssuesScreen(props: IIssuesScreenProps) {
  const {
    list: issues,
    loading: loadingIssues,
    filters,
    sortCriteria,
    organizationSlug,
    repoSlug,
    page,
    error,
  } = useSelector((state: IApplicationState) => state.issuesReducer);
  const { list: bookmarks, loading: loadingBookmarks } = useSelector(
    (state: IApplicationState) => state.bookmarksReducer
  );

  const dispatch = useDispatch();
  const setOrganizationSlug = (slug: string): void => {
    dispatch(issueActions.setOrganizationSlug(slug));
  };
  const setRepoSlug = (slug: string): void => {
    dispatch(issueActions.setRepoSlug(slug));
  };
  const goToNextPage = (): void => {
    dispatch(issueActions.setPage(page + 1));
    getIssues();
  };
  const goToPreviousPage = (): void => {
    dispatch(issueActions.setPage(page - 1));
    getIssues();
  };
  const getIssues = (): void => {
    dispatch(issueActions.getIssues());
    collapseRepoPicker();
  };

  const toggleFilter = (filterId: string): void => {
    LayoutAnimation.easeInEaseOut();
    dispatch(issueActions.toggleFilter(filterId));
    getIssues();
  };

  const showSortOptions = (): void => {
    ActionSheetIOS.showActionSheetWithOptions(
      {
        options: ['Cancel', ...sortCriteria.map((x) => x.label)],
        cancelButtonIndex: 0,
      },
      (buttonIndex) => {
        if (buttonIndex > 0) {
          LayoutAnimation.easeInEaseOut();
          dispatch(issueActions.setSortCriterion(sortCriteria[buttonIndex - 1].id));
          getIssues();
        }
      }
    );
  };

  useEffect(() => {
    getIssues();
    dispatch(bookmarkActions.getBookmarks());
  }, []);

  const [isExpanded, setIsExpanded] = useState(false);

  const expandRepoPicker = () => {
    LayoutAnimation.easeInEaseOut();
    setIsExpanded(true);
  };

  const collapseRepoPicker = () => {
    LayoutAnimation.easeInEaseOut();
    setIsExpanded(false);
  };

  return (
    <View style={styles.container}>
      {isExpanded ? (
        <View style={styles.card}>
          <View style={styles.repositoryPicker}>
            <TextInput halfWidth name="Organization" value={organizationSlug} onChangeText={setOrganizationSlug} />
            <Text style={{ marginTop: 26, fontSize: 20, marginLeft: 2, color: Color.border }}>/</Text>
            <TextInput halfWidth name="Repository" value={repoSlug} onChangeText={setRepoSlug} />
          </View>
          <View style={styles.buttonsWrapper}>
            <Button type="secondary" text="Cancel" onPress={collapseRepoPicker} style={{ width: 130 }} />
            <Button
              type="quaternary"
              text="View issues"
              leftIcon={<AntDesign size={20} color={Color.blue} name="github" style={{ marginRight: 6 }} />}
              onPress={getIssues}
            />
          </View>

          {loadingIssues ||
            (loadingBookmarks && (
              <View style={styles.loading}>
                <ActivityIndicator color={Color.blue} />
              </View>
            ))}
        </View>
      ) : (
        <View style={[styles.spacedRow, { marginTop: 0 }]}>
          <Feather size={20} name="git-branch" color={Color.border} style={styles.filterIcon} />
          <Text style={styles.repoText}>
            {organizationSlug} / {repoSlug}
          </Text>
          <TouchableOpacity onPress={expandRepoPicker} style={styles.editRepoButton}>
            <Feather size={14} name="edit-2" color={Color.blue} />
          </TouchableOpacity>
        </View>
      )}

      <View style={styles.spacedRow}>
        <Feather size={20} name="sliders" color={Color.border} style={styles.filterIcon} />
        {filters.map(({ id, label, isActive }) => (
          <TouchableOpacity
            onPress={() => toggleFilter(id)}
            activeOpacity={1}
            key={id}
            style={[styles.filter, isActive ? styles.activeFilter : {}]}
          >
            {isActive && (
              <Feather size={16} name="check" color={Color.blue} style={{ marginRight: 4, marginLeft: -4 }} />
            )}
            <Text style={{ color: Color.blue }}>{label} issues</Text>
          </TouchableOpacity>
        ))}
      </View>
      <View style={styles.spacedRow}>
        <Feather size={20} name="arrow-up" color={Color.border} style={styles.filterIcon} />
        <TouchableOpacity onPress={showSortOptions} style={styles.sortButton}>
          <FontAwesome size={17} name="caret-down" color={Color.blue} style={styles.caretIcon} />
          <Text style={{ color: Color.blue }}>Sort By {sortCriteria.find((x) => x.isActive)?.label}</Text>
        </TouchableOpacity>
        {(loadingIssues || loadingBookmarks) && (
          <View style={styles.loading}>
            <ActivityIndicator color={Color.blue} />
          </View>
        )}
      </View>

      {error && <Text style={styles.error}>An error occurred: {error.message}</Text>}

      <View style={styles.issues}>
        <FlatList
          data={issues}
          keyExtractor={(item) => item.id + ''}
          contentContainerStyle={{ paddingHorizontal: 16, paddingBottom: 150 }}
          renderItem={({ item }) => (
            <Issue
              issue={{ ...item, isBookmarked: bookmarks.some((x) => x.id === item.id) }}
              navigation={props.navigation}
            />
          )}
        />
      </View>

      <View pointerEvents="box-none" style={styles.paginationButtons}>
        <Button
          style={{ width: 120 }}
          disabled={page < 2}
          type="quaternary"
          text="Previous"
          leftIcon={<Feather size={20} name="chevron-left" color={page < 2 ? Color.border : Color.blue} />}
          onPress={goToPreviousPage}
        />
        <Button
          text="Next"
          type="quaternary"
          style={{ width: 120 }}
          rightIcon={<Feather size={20} name="chevron-right" color={Color.blue} />}
          onPress={goToNextPage}
        />
        <LinearGradient
          pointerEvents="none"
          colors={[Color.steel + '00', Color.steel + '44']}
          style={styles.gradient}
        />
      </View>
    </View>
  );
}
