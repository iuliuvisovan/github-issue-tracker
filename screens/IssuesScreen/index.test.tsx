import React from 'react';
import IssuesScreen from './index';
import MockedNavigator from '../../navigation/MockedNavigator';
import { create, act, ReactTestRenderer } from 'react-test-renderer';
import { Provider as ReduxProvider } from 'react-redux';
import store from '../../redux';
import mockIssues from '../../mocks/issues';
import fetchMock from 'fetch-mock';
import isDarkColor from 'is-dark-color';
import { NativeSyntheticEvent, TextInputFocusEventData, TextInputProps } from 'react-native';
import { TextInput } from 'react-native-gesture-handler';
import { Color } from '../../components';

jest.mock('is-dark-color');
isDarkColor.mockReturnValue(true);

const createFreshTree = () =>
  create(
    <ReduxProvider store={store}>
      <MockedNavigator component={IssuesScreen} />
    </ReduxProvider>
  );

describe('Issues Screen', () => {
  afterEach(() => {
    fetchMock.restore();
  });

  it('renders correctly', async () => {
    let tree;

    fetchMock.getOnce('https://api.github.com/repos/facebook/react-native/issues?state=all&sort=created&page=1', {
      body: mockIssues,
      headers: { 'content-type': 'application/json' },
    });

    await act(async () => {
      tree = createFreshTree();
    });

    expect(tree).toMatchSnapshot();
  });

  it('toggles filter correctly', async () => {
    let tree: ReactTestRenderer;

    fetchMock.getOnce('https://api.github.com/repos/facebook/react-native/issues?state=all&sort=created&page=1', {
      body: mockIssues,
      headers: { 'content-type': 'application/json' },
    });
    fetchMock.getOnce('https://api.github.com/repos/facebook/react-native/issues?state=closed&sort=created&page=1', {
      body: mockIssues,
      headers: { 'content-type': 'application/json' },
    });

    await act(async () => {
      tree = createFreshTree();
    });

    const openFilter = tree.root.findByProps({ testID: 'open' }).props;

    await act(async () => openFilter.onPress());

    expect(tree).toMatchSnapshot();
  });

  it('expands RepoPicker correctly', async () => {
    let tree: ReactTestRenderer;

    fetchMock.getOnce('https://api.github.com/repos/facebook/react-native/issues?state=closed&sort=created&page=1', {
      body: mockIssues,
      headers: { 'content-type': 'application/json' },
    });

    await act(async () => {
      tree = createFreshTree();
    });

    const editButton = tree.root.findByProps({ testID: 'expandPickerButton' }).props;

    await act(async () => editButton.onPress());

    const organizationInput = tree.root.findByProps({ testID: 'organizationInput' });

    const innerTextInput = organizationInput.findByType('TextInput' as any);

    await act(async () => innerTextInput.props.onFocus({} as NativeSyntheticEvent<TextInputFocusEventData>));

    expect(innerTextInput.props.style[1].borderColor).toEqual(Color.blue);

    await act(async () => innerTextInput.props.onBlur({} as NativeSyntheticEvent<TextInputFocusEventData>));

    expect(innerTextInput.props.style[1].borderColor).toEqual(Color.border);
  });
});
