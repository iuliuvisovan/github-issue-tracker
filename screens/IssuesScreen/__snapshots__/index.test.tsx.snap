// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      pointerEvents="box-none"
      style={
        Object {
          "zIndex": 1,
        }
      }
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        onLayout={[Function]}
        pointerEvents="box-none"
        style={null}
      >
        <View
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "zIndex": 0,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "rgb(255, 255, 255)",
                "borderBottomColor": "rgb(216, 216, 216)",
                "flex": 1,
                "shadowColor": "rgb(216, 216, 216)",
                "shadowOffset": Object {
                  "height": 0.5,
                  "width": 0,
                },
                "shadowOpacity": 0.85,
                "shadowRadius": 0,
              }
            }
          />
        </View>
        <View
          pointerEvents="box-none"
          style={
            Object {
              "height": 64,
              "maxHeight": undefined,
              "minHeight": undefined,
              "opacity": undefined,
              "transform": undefined,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "height": 20,
              }
            }
          />
          <View
            pointerEvents="box-none"
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "marginHorizontal": 16,
                  "opacity": 1,
                }
              }
            >
              <Text
                accessibilityRole="header"
                aria-level="1"
                numberOfLines={1}
                onLayout={[Function]}
                style={
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 17,
                    "fontWeight": "600",
                  }
                }
              >
                MockedScreen
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "opacity": 1,
            }
          }
        />
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onGestureEnd={[Function]}
          onOpen={[Function]}
          onTransition={[Function]}
          pointerEvents="box-none"
          style={
            Array [
              Object {
                "overflow": undefined,
              },
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
            ]
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            needsOffscreenAlphaCompositing={false}
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              collapsable={false}
              forwardedRef={[Function]}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                Object {
                  "flex": 1,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "rgb(242, 242, 242)",
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": -1,
                    },
                    "shadowOpacity": 0.3,
                    "shadowRadius": 5,
                    "top": 0,
                    "width": 3,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "overflow": "hidden",
                    },
                    Array [
                      Object {
                        "backgroundColor": "rgb(242, 242, 242)",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "column-reverse",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "backgroundColor": "#f4f6f6",
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "backgroundColor": "#f4f6f6",
                              "padding": 16,
                            },
                            Object {},
                          ]
                        }
                      >
                        <View
                          style={
                            Array [
                              Object {
                                "alignItems": "center",
                                "flexDirection": "row",
                                "marginTop": 16,
                              },
                              Object {
                                "marginTop": 0,
                              },
                            ]
                          }
                        >
                          <Text
                            allowFontScaling={false}
                            style={
                              Array [
                                Object {
                                  "color": "#aab2ba",
                                  "fontSize": 20,
                                },
                                Object {
                                  "marginLeft": 8,
                                  "marginRight": 8,
                                  "marginTop": 2,
                                },
                                Object {
                                  "fontFamily": "feather",
                                  "fontStyle": "normal",
                                  "fontWeight": "normal",
                                },
                                Object {},
                              ]
                            }
                          >
                            ï…¶
                          </Text>
                          <Text
                            style={
                              Array [
                                Object {
                                  "color": "#14181c",
                                  "fontFamily": "muli",
                                  "lineHeight": 16,
                                },
                                Object {
                                  "color": "#414d5b",
                                  "fontSize": 16,
                                  "marginLeft": 8,
                                  "marginTop": 6,
                                },
                              ]
                            }
                          >
                            facebook / react-native
                          </Text>
                          <View
                            accessible={true}
                            focusable={true}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "alignItems": "center",
                                "backgroundColor": "#fff",
                                "borderColor": "#00aaff",
                                "borderRadius": 16,
                                "borderWidth": 1,
                                "height": 30,
                                "justifyContent": "center",
                                "marginLeft": 12,
                                "opacity": 1,
                                "paddingLeft": 1,
                                "paddingTop": 0,
                                "width": 30,
                              }
                            }
                          >
                            <Text
                              allowFontScaling={false}
                              style={
                                Array [
                                  Object {
                                    "color": "#00aaff",
                                    "fontSize": 14,
                                  },
                                  undefined,
                                  Object {
                                    "fontFamily": "feather",
                                    "fontStyle": "normal",
                                    "fontWeight": "normal",
                                  },
                                  Object {},
                                ]
                              }
                            >
                              ï… 
                            </Text>
                          </View>
                        </View>
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "flexDirection": "row",
                              "marginTop": 16,
                            }
                          }
                        >
                          <Text
                            allowFontScaling={false}
                            style={
                              Array [
                                Object {
                                  "color": "#aab2ba",
                                  "fontSize": 20,
                                },
                                Object {
                                  "marginLeft": 8,
                                  "marginRight": 8,
                                  "marginTop": 2,
                                },
                                Object {
                                  "fontFamily": "feather",
                                  "fontStyle": "normal",
                                  "fontWeight": "normal",
                                },
                                Object {},
                              ]
                            }
                          >
                            ï‡ 
                          </Text>
                          <View
                            accessible={true}
                            focusable={true}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "alignItems": "center",
                                "backgroundColor": "#fff",
                                "borderColor": "#00aaff",
                                "borderRadius": 16,
                                "borderWidth": 1,
                                "flexDirection": "row",
                                "height": 30,
                                "marginLeft": 8,
                                "opacity": 1,
                                "paddingHorizontal": 16,
                                "paddingTop": 6,
                                "paddingVertical": 4,
                              }
                            }
                          >
                            <Text
                              allowFontScaling={false}
                              style={
                                Array [
                                  Object {
                                    "color": "#00aaff",
                                    "fontSize": 16,
                                  },
                                  Object {
                                    "marginLeft": -4,
                                    "marginRight": 4,
                                  },
                                  Object {
                                    "fontFamily": "feather",
                                    "fontStyle": "normal",
                                    "fontWeight": "normal",
                                  },
                                  Object {},
                                ]
                              }
                            >
                              ï„«
                            </Text>
                            <Text
                              style={
                                Array [
                                  Object {
                                    "color": "#14181c",
                                    "fontFamily": "muli",
                                    "lineHeight": 16,
                                  },
                                  Object {
                                    "color": "#00aaff",
                                  },
                                ]
                              }
                            >
                              Open
                               issues
                            </Text>
                          </View>
                          <View
                            accessible={true}
                            focusable={true}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "alignItems": "center",
                                "backgroundColor": "#fff",
                                "borderColor": "#00aaff",
                                "borderRadius": 16,
                                "borderWidth": 1,
                                "flexDirection": "row",
                                "height": 30,
                                "marginLeft": 8,
                                "opacity": 1,
                                "paddingHorizontal": 16,
                                "paddingTop": 6,
                                "paddingVertical": 4,
                              }
                            }
                          >
                            <Text
                              allowFontScaling={false}
                              style={
                                Array [
                                  Object {
                                    "color": "#00aaff",
                                    "fontSize": 16,
                                  },
                                  Object {
                                    "marginLeft": -4,
                                    "marginRight": 4,
                                  },
                                  Object {
                                    "fontFamily": "feather",
                                    "fontStyle": "normal",
                                    "fontWeight": "normal",
                                  },
                                  Object {},
                                ]
                              }
                            >
                              ï„«
                            </Text>
                            <Text
                              style={
                                Array [
                                  Object {
                                    "color": "#14181c",
                                    "fontFamily": "muli",
                                    "lineHeight": 16,
                                  },
                                  Object {
                                    "color": "#00aaff",
                                  },
                                ]
                              }
                            >
                              Closed
                               issues
                            </Text>
                          </View>
                        </View>
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "flexDirection": "row",
                              "marginTop": 16,
                            }
                          }
                        >
                          <Text
                            allowFontScaling={false}
                            style={
                              Array [
                                Object {
                                  "color": "#aab2ba",
                                  "fontSize": 20,
                                },
                                Object {
                                  "marginLeft": 8,
                                  "marginRight": 8,
                                  "marginTop": 2,
                                },
                                Object {
                                  "fontFamily": "feather",
                                  "fontStyle": "normal",
                                  "fontWeight": "normal",
                                },
                                Object {},
                              ]
                            }
                          >
                            ï„”
                          </Text>
                          <View
                            accessible={true}
                            focusable={true}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "alignItems": "center",
                                "backgroundColor": "#fff",
                                "borderColor": "#00aaff",
                                "borderRadius": 16,
                                "borderWidth": 1,
                                "flexDirection": "row",
                                "height": 30,
                                "marginLeft": 8,
                                "opacity": 1,
                                "paddingHorizontal": 16,
                                "paddingTop": 2,
                              }
                            }
                          >
                            <Text
                              allowFontScaling={false}
                              style={
                                Array [
                                  Object {
                                    "color": "#00aaff",
                                    "fontSize": 17,
                                  },
                                  Object {
                                    "marginLeft": -2,
                                    "marginRight": 8,
                                    "marginTop": -1,
                                  },
                                  Object {
                                    "fontFamily": "FontAwesome",
                                    "fontStyle": "normal",
                                    "fontWeight": "normal",
                                  },
                                  Object {},
                                ]
                              }
                            >
                              ïƒ—
                            </Text>
                            <Text
                              style={
                                Array [
                                  Object {
                                    "color": "#14181c",
                                    "fontFamily": "muli",
                                    "lineHeight": 16,
                                  },
                                  Object {
                                    "color": "#00aaff",
                                  },
                                ]
                              }
                            >
                              Sort By 
                              Created Date
                            </Text>
                          </View>
                        </View>
                      </View>
                      <Text
                        style={
                          Array [
                            Object {
                              "color": "#14181c",
                              "fontFamily": "muli",
                              "lineHeight": 16,
                            },
                            Object {
                              "alignSelf": "center",
                              "color": "#ea1f59",
                              "marginTop": 16,
                            },
                          ]
                        }
                      >
                        An error occurred: 
                        fetch is not defined
                      </Text>
                      <RCTScrollView
                        contentContainerStyle={
                          Object {
                            "paddingBottom": 80,
                            "paddingHorizontal": 16,
                            "paddingTop": 16,
                          }
                        }
                        data={Array []}
                        disableVirtualization={false}
                        getItem={[Function]}
                        getItemCount={[Function]}
                        horizontal={false}
                        initialNumToRender={10}
                        keyExtractor={[Function]}
                        maxToRenderPerBatch={10}
                        onContentSizeChange={[Function]}
                        onEndReachedThreshold={2}
                        onLayout={[Function]}
                        onMomentumScrollEnd={[Function]}
                        onScroll={[Function]}
                        onScrollBeginDrag={[Function]}
                        onScrollEndDrag={[Function]}
                        removeClippedSubviews={false}
                        renderItem={[Function]}
                        scrollEventThrottle={50}
                        stickyHeaderIndices={Array []}
                        updateCellsBatchingPeriod={50}
                        viewabilityConfigCallbackPairs={Array []}
                        windowSize={21}
                      >
                        <View />
                      </RCTScrollView>
                      <View
                        pointerEvents="box-none"
                        style={
                          Object {
                            "bottom": 0,
                            "flexDirection": "row",
                            "justifyContent": "space-between",
                            "left": 0,
                            "paddingHorizontal": 16,
                            "position": "absolute",
                            "width": 750,
                          }
                        }
                      >
                        <View
                          accessible={true}
                          focusable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "#fff",
                              "borderColor": "#aab2ba",
                              "borderRadius": 609,
                              "borderWidth": 1,
                              "elevation": 0,
                              "flex": 0,
                              "flexDirection": "row",
                              "height": 39,
                              "justifyContent": "center",
                              "marginVertical": 8,
                              "opacity": 1,
                              "paddingHorizontal": 16,
                              "shadowColor": "#aab2ba",
                              "shadowOffset": Object {
                                "height": 2,
                                "width": 0,
                              },
                              "shadowOpacity": 0,
                              "shadowRadius": 4,
                              "width": 120,
                            }
                          }
                        >
                          <Text
                            allowFontScaling={false}
                            style={
                              Array [
                                Object {
                                  "color": "#aab2ba",
                                  "fontSize": 20,
                                },
                                undefined,
                                Object {
                                  "fontFamily": "feather",
                                  "fontStyle": "normal",
                                  "fontWeight": "normal",
                                },
                                Object {},
                              ]
                            }
                          >
                            ï„¯
                          </Text>
                          <Text
                            style={
                              Array [
                                Object {
                                  "color": "#14181c",
                                  "fontFamily": "muli",
                                  "lineHeight": 16,
                                },
                                Object {
                                  "color": "#00aaff",
                                  "fontFamily": "muli-extra-bold",
                                  "fontSize": 12,
                                  "letterSpacing": 0.8,
                                },
                                Object {
                                  "color": "#00aaff",
                                  "marginHorizontal": 2,
                                },
                                Object {
                                  "color": "#aab2ba",
                                },
                              ]
                            }
                          >
                            PREVIOUS
                          </Text>
                        </View>
                        <View
                          accessible={true}
                          focusable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "#fff",
                              "borderColor": "#00aaff",
                              "borderRadius": 609,
                              "borderWidth": 1,
                              "elevation": 0,
                              "flex": 0,
                              "flexDirection": "row",
                              "height": 39,
                              "justifyContent": "center",
                              "marginVertical": 8,
                              "opacity": 1,
                              "paddingHorizontal": 16,
                              "shadowColor": "#aab2ba",
                              "shadowOffset": Object {
                                "height": 2,
                                "width": 0,
                              },
                              "shadowOpacity": 0,
                              "shadowRadius": 4,
                              "width": 120,
                            }
                          }
                        >
                          <Text
                            style={
                              Array [
                                Object {
                                  "color": "#14181c",
                                  "fontFamily": "muli",
                                  "lineHeight": 16,
                                },
                                Object {
                                  "color": "#00aaff",
                                  "fontFamily": "muli-extra-bold",
                                  "fontSize": 12,
                                  "letterSpacing": 0.8,
                                },
                                Object {
                                  "color": "#00aaff",
                                  "marginHorizontal": 2,
                                },
                                Object {},
                              ]
                            }
                          >
                            NEXT
                          </Text>
                          <Text
                            allowFontScaling={false}
                            style={
                              Array [
                                Object {
                                  "color": "#00aaff",
                                  "fontSize": 20,
                                },
                                undefined,
                                Object {
                                  "fontFamily": "feather",
                                  "fontStyle": "normal",
                                  "fontWeight": "normal",
                                },
                                Object {},
                              ]
                            }
                          >
                            ï„°
                          </Text>
                        </View>
                        <ViewManagerAdapter_ExpoLinearGradient
                          pointerEvents="none"
                          proxiedProperties={
                            Object {
                              "colors": Array [
                                4279643,
                                1145130331,
                              ],
                              "endPoint": undefined,
                              "locations": undefined,
                              "startPoint": undefined,
                            }
                          }
                          style={
                            Object {
                              "bottom": 0,
                              "height": 40,
                              "left": 0,
                              "position": "absolute",
                              "width": 750,
                              "zIndex": -1,
                            }
                          }
                        />
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaProvider>
</View>
`;
